substrate/frame/support/src/traits/tokens/currency/lockable.rs

```diff
@@ -112,3 +112,52 @@ pub trait VestingSchedule<AccountId> {
 	/// NOTE: This doesn't alter the free balance of the account.
 	fn remove_vesting_schedule(who: &AccountId, schedule_index: u32) -> DispatchResult;
 }
+
+/// A vested transfer over a currency. This allows a transferred amount to vest over time.
+pub trait VestedTransfer<AccountId> {
+	/// The quantity used to denote time; usually just a `BlockNumber`.
+	type Moment;
+
+	/// The currency that this schedule applies to.
+	type Currency: Currency<AccountId>;
```

Comment:
I see a lot of PR moving from Currency to Fungible.
What are the reason to use one or the other? here we could use Fungible with Hold I guess.

Reply:
Ideally we would change to fungible, but it would not be a trivial change due to the way that Freezes/Holds are managed versus the Currency Trait's Reserve/Locks.

It would really be a much bigger PR, and I don't think I have the energy for that one right now.

Reply:
Good to know, yes it should be done in another PR if any